common/common.go

```diff
@@ -79,6 +80,16 @@ var (
 	errHTTPClientInvalid       = errors.New("custom http client cannot be nil")
 )
 
+// NilGuard returns an ErrNilPointer with the type of the first nil argument
+func NilGuard(ptrs ...any) (errs error) {
+	for _, p := range ptrs {
+		if (*[2]uintptr)(unsafe.Pointer(&p))[1] == 0 {
+			errs = AppendError(errs, fmt.Errorf("%w: %T", ErrNilPointer, p))
+		}
```

Comment:
This could be much cleaner like [this](https://go.dev/play/p/5WG6e9R0hOx)
```
type eface struct {
	_type unsafe.Pointer // *abi.Type (golang.org/internal/abi)
	addr  unsafe.Pointer
}

e := (*eface)(unsafe.Pointer(&p))
if e.addr == nil {
}
```
But instead I'm choosing to add a lot of comments to clarify, and keep the efficiency of checking uintptrs directly.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/2bf8646611

Reply:
I appreciate the comments. I hadn't yet looked into the weird workings of the double casting slice check